cdc/log.cc

```diff
@@ -548,22 +555,29 @@ class transformer final {
                             if (pirow && pirow->has(cdef.name_as_text())) {
                                 values[0] = data_type_for<column_op_native_type>()->decompose(data_value(static_cast<column_op_native_type>(column_op::set)));
                                 values[1] = pirow->get_blob(cdef.name_as_text());
-                                values[2] = std::nullopt;
 
                                 assert(std::addressof(res.partition().clustered_row(*_log_schema, *pikey)) != std::addressof(res.partition().clustered_row(*_log_schema, log_ck)));
                                 assert(pikey->explode() != log_ck.explode());
                                 res.set_cell(*pikey, *dst, atomic_cell::make_live(*dst->type, ts, tuple_type_impl::build_value(values)));
                             }
+
                         } else {
                             cdc_log.warn("Non-atomic cell ignored {}.{}:{}", _schema->ks_name(), _schema->cf_name(), cdef.name_as_text());
                         }
                     });
                 };
 
+                const auto& marker = r.row().marker();
+                if (marker.is_expiring()) {
```

Comment:
The comment on `row_marker::is_expiring()` says:
```
    // Can be called only when is_live().
```
but there are cases here when `!marker.is_live()`, aren't there?

Reply:
Added a check for liveness.